### ISSUE TITLE
Unspecified Div Op behavior for negative quotients

### ISSUE THREAD

**Issue by yuanyao-nv** [onnx/onnx#4016]
Documentation for the div operator currently does not provide any info on its behavior for negative ints. 
For negative quotients, 
- the Python `//` operator rounds away from zero, ie `-11//3` gives `-4`
- both ONNX runtime and TensorRT (`kDiv`) are defaulting to rounding towards zero, ie `-11//3` gives `-3`.

Please provide some clarity on the spec. Thank you!

**Comment by diyessi** [comment]
Since most models originate in Python, I think the Python behavior makes the most sense. It would be interesting to know if the ONNX Runtime and TensorRT behavior was accidental; a lot of people aren't aware that C and Python do different things (I think it used to be unspecified behavior in C).

**Comment by garymm** [comment]
Someone needs to investigate how this is handled in:
* PyTorch 
* TensorFlow
* ONNX Runtime: CPU EP (x86 and ARM), CUDA EP, ROCm EP
* TensorRT

Then we can use that info to make a proposed change to the spec.

@yuanyao-nv if you can do this research that would help us make a decision on this.

**Comment by jcwchen** [comment]
Provide some findings here:

- [TensorFlow](https://github.com/tensorflow/tensorflow/blob/v2.8.0/tensorflow/python/ops/math_ops.py#L445-L477): Basically tensorflow.divide uses Python's `/` and the result will always be float without any round even if the inputs are integer. For instance, -11/3=-3.6666667.
- [PyTorch](https://pytorch.org/docs/stable/generated/torch.div.html#torch.div): Similar to TensorFlow. By default, `torch.div` produces float result, but it does have attribute "rounding_mode" to represent None (without any round), trunc, floor.

It seems that both TensorFlow and PyTorch follow Python's behavior (`/`). Besides, PyTorch further has attribute to represent different round method, which might be a good solution to prevent confusion about difference between Python div and C++ div.

**Comment by diyessi** [comment]
We used Python semantics in ngraph.
I think I have seen shape computations that depended on it.

**Comment by gramalingam** [comment]
I agree we need to clarify the spec. (By the way, I think that users can achieve the python `//` behavior by casting to float, performing the computation, and then computing a `floor`. Perhaps the converters could use such a translation in case we settle on the `trunc` behavior.)

**Comment by xenova** [comment]
ran into this today, when trying to replace a ceil operation with an equivalent version (to keep operands as ints)

replacing
```
num_blocks = math.ceil(num_features / self.context_size)
```
with (this one works)

```
nblocks = (seq_len + self.window_size - 1) // self.window_size
```

vs. (breaks)
```
nblocks = -(seq_len // -self.window_size)
```

---

Running with ONNXRuntime produced different results from pytorch export.

**Comment by LeonxLJX** [comment]
I'd like to take this one.

From the report, the useful first step is to pin down the exact code path involved before touching anything — these tend to have a small true cause behind a broad-looking symptom, and guessing at it usually produces a patch that masks the behavior instead of fixing it.

Plan: trace the path, reproduce it, then make a focused change with a test if this area has coverage. I'll keep the diff small so it's easy to review.

If there's an existing convention in this repo for how this kind of thing is handled, I'd rather follow it than invent my own — happy to be pointed at one.

(claiming via @LeonxLJX)

**Comment by migarci2** [comment]
Opened a docs clarification PR that adds an explicit negative-quotient example for integer `Div` (truncating / toward zero, e.g. `-11 / 3` → `-3`, matching ORT/TRT): https://github.com/onnx/onnx/pull/8461
